libs/wingc/src/type_check.rs

```diff
@@ -1451,16 +1460,16 @@ impl<'a> TypeChecker<'a> {
 					.as_struct()
 					.expect(&format!("Expected \"{}\" to be a struct type", struct_type));
 
-				// Verify that all fields are present and are of the right type
+				// Verify that all expected fields are present and are the right type
 				for (name, kind, _info) in st.env.iter(true) {
 					let field_type = kind
 						.as_variable()
 						.expect("Expected struct field to be a variable in the struct env")
 						.type_;
-					match fields.get(&name) {
-						Some((_, field_exp)) => {
-							let t = self.type_check_exp(field_exp, env, statement_idx, context);
-							self.validate_type(t, field_type, field_exp);
+					match fields.get(&Symbol::global(&name)) {
```

Comment:
Do we want to go as far as implementing [Equivalent](https://docs.rs/indexmap/latest/indexmap/trait.Equivalent.html) for Symbol so we can do this with a String?

Reply:
oh, interesting. Probably a good idea